services/src/main/java/org/jboss/windup/web/services/rest/WindupEndpointImpl.java

```diff
@@ -55,18 +65,49 @@
     private Queue executorQueue;
 
     /**
+     * To avoid problems with passing detached entities,
+     * load related entities during cloning
+     */
+    protected AnalysisContext cloneExecution(AnalysisContext originalContext)
+    {
+        Set<Package> allPackages = new HashSet<>();
+
+        allPackages.addAll(originalContext.getIncludePackages());
+        allPackages.addAll(originalContext.getExcludePackages());
+
+        List<Long> packageIds = allPackages.stream().map(aPackage -> aPackage.getId())
+                .collect(Collectors.toList());
+
+        AnalysisContext clone = originalContext.clone();
```

Comment:
That's right... I was thinking about using version and throwing an exception if there is mismatch too. If it is an acceptable solution, then I think it is the simplest one.